ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -29,4 +31,40 @@ message Completion {
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Optional
   string transaction_id = 3;
+
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string application_id = 4;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of ``party`` and ``act_as`` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  repeated string act_as = 5;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string submission_id = 6;
+
+  // The actual deduplication window used for the submission, which is derived from
+  // ``Commands.deduplication_period``. The ledger may convert the deduplication period into other
+  // descriptions and extend the period in implementation-specified ways.
+  //
+  // Used to audit the deduplication guarantee described in ``commands.proto``.
+  //
+  // Optional; the deduplication guarantee applies even if the completion omits this field.
+  oneof deduplication_period {
+    // Specifies the length of the deduplication period.
+    // It is interpreted relative to the local clock at some point during the submission's processing.
+    //
+    // Must be non-negative.
+    google.protobuf.Duration deduplication_time = 11;
+
+    // Specifies a point in time in the past after which this submission should be de-duplicated w.r.t. previous submissions of the same commands.
+    // It is interpreted relative to the local clock at some point during the submission's processing.
+    google.protobuf.Timestamp deduplication_start = 12;
```

Comment:
What has happened to the `deduplication_offset`?

Currently, `DeduplicationPeriod` in `state.v2` offers only `DeduplicationDuration` and `DeduplicationOffset`. Why do we have in completions now duration and timestamp, but not start?

Reply:
Didn't we agree that we'd not worry about deduplication offsets for now?

Reply:
No. We agreed on not worrying about deduplication starting points for now.

Reply:
Hmmm, there was clearly a miscommunication here. Let's take this to a different channel.

Reply:
Added 89336ab